src/rust/engine/graph/src/lib.rs

```diff
@@ -591,16 +597,19 @@ impl<N: Node> Graph<N> {
   ) where
     C: NodeContext<Node = N>,
   {
-    let mut inner = self.inner.lock().unwrap();
-    // Get the Generations of all dependencies of the Node. We can trust that these have not changed
-    // since we began executing, as long as we are not currently marked dirty (see the method doc).
-    let dep_generations = inner
-      .pg
-      .neighbors_directed(entry_id, Direction::Outgoing)
-      .filter_map(|dep_id| inner.entry_for_id(dep_id))
-      .map(|entry| entry.generation())
-      .collect();
-    if let Some(entry) = inner.entry_for_id_mut(entry_id) {
+    let (entry, entry_id, dep_generations) = {
+      let mut inner = self.inner.lock().unwrap();
+      // Get the Generations of all dependencies of the Node. We can trust that these have not changed
+      // since we began executing, as long as we are not currently marked dirty (see the method doc).
```

Comment:
Is this still true? Hm.

I believe that this is still true, as long as we continue to ensure that Node clearing/dirtying occurs under the graph lock. If that sounds right (... I only spent 5 minutes thinking about it), could you update the comments here and there?